gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/resource/SubscriptionResource.java

```diff
@@ -97,20 +117,60 @@ public Response getSubscriptionBySubscriptionId(
     public Response closeSubscription(@PathParam("subscriptionId") String subscriptionId) {
         SubscriptionEntity subscriptionEntity = subscriptionService.findById(subscriptionId);
         final ExecutionContext executionContext = GraviteeContext.getExecutionContext();
-        if (
-            hasPermission(
-                executionContext,
-                RolePermission.APPLICATION_SUBSCRIPTION,
-                subscriptionEntity.getApplication(),
-                RolePermissionAction.DELETE
-            )
-        ) {
+        if (hasPermission(executionContext, APPLICATION_SUBSCRIPTION, subscriptionEntity.getApplication(), RolePermissionAction.DELETE)) {
             subscriptionService.close(executionContext, subscriptionId);
             return Response.noContent().build();
         }
         throw new ForbiddenAccessException();
     }
 
+    @PUT
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response updateSubscription(
+        @PathParam("subscriptionId") String subscriptionId,
+        @Valid @NotNull SubscriptionInput subscriptionInput
+    ) {
+        final ExecutionContext executionContext = GraviteeContext.getExecutionContext();
+
+        SubscriptionEntity subscriptionEntity = subscriptionService.findById(subscriptionId);
+
+        if (!hasPermission(executionContext, APPLICATION_SUBSCRIPTION, subscriptionEntity.getApplication(), UPDATE)) {
+            throw new ForbiddenAccessException();
+        }
+
+        if (subscriptionEntity.getStatus() == CLOSED) {
+            throw new SubscriptionNotUpdatableException(subscriptionId);
+        }
+
+        GenericPlanEntity planEntity = planSearchService.findById(executionContext, subscriptionEntity.getPlan());
+
+        UpdateSubscriptionEntity updateSubscriptionEntity = new UpdateSubscriptionEntity();
+        updateSubscriptionEntity.setId(subscriptionId);
+        updateSubscriptionEntity.setEndingAt(subscriptionEntity.getEndingAt());
+        updateSubscriptionEntity.setStartingAt(subscriptionEntity.getStartingAt());
+        if (subscriptionInput.getConfiguration() != null) {
+            updateSubscriptionEntity.setConfiguration(MAPPER.valueToTree(subscriptionInput.getConfiguration()));
+        }
+        if (subscriptionInput.getFilter() != null) {
+            updateSubscriptionEntity.setFilter(subscriptionInput.getFilter());
+        }
+        if (subscriptionInput.getMetadata() != null) {
+            updateSubscriptionEntity.setMetadata(subscriptionInput.getMetadata());
+        }
+
+        io.gravitee.repository.management.model.Subscription.Status newSubscriptionStatus = planEntity.getPlanValidation() == MANUAL
```

Comment:
Looks like a business rule which should not be there but in the service instead. WDYT ?

Reply:
As discussed together, this rule is now on service side.
I used dedicated objects for subscription configuration update by consumer, as he can only update configuration and filter ; And so a dedicated service method